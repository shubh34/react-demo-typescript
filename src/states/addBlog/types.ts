export interface IAddBlogState {
    title: string,
    categories: string[],
    content: string,
}
