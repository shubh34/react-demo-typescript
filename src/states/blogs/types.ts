import { Moment } from 'moment'
export interface IBlog {
    id ? : string,
    categories: string [],
    date ?: Moment | string,
    content: string,
    title: string,

}
export interface IBlogsState {
    blogs: IBlog [],
}