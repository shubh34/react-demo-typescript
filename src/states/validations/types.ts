export interface IForm {
    title: Object,
    categories: Object,
    content: Object,

}
export interface IValidationState {
    form: IForm | any,
}