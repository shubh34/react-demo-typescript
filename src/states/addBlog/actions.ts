
export const INIT_ADD_BLOG: string = "INIT_ADD_BLOG";
export const UPDATE_FORM_FIELD: string = "UPDATE_FORM_FIELD";
export const ADD_BLOG_CATEGORY: string = "ADD_BLOG_CATEGORY";
export const DELETE_BLOG_CATEGORY: string = "DELETE_BLOG_CATEGORY";

export const initAddBlogForm = () => {
    return {
        type: INIT_ADD_BLOG,
    }
}

export const updateFormField = (name: string, value: string) => {
    return {
        type: UPDATE_FORM_FIELD,
        field: {
            name,
            value,
        }
    }
}

export const addCategory = (category: string) => {
    return {
        type: ADD_BLOG_CATEGORY,
        category,
    }
}

export const deleteCategory = (category: string) => {
    return {
        type: DELETE_BLOG_CATEGORY,
        category,
    }
}


