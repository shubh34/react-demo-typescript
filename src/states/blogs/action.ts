import uuid from 'node-uuid';
import moment from 'moment';
import { IBlog } from "./types";

export const ADD_BLOG: string = "ADD_BLOG";
export const EDIT_BLOG: string = "EDIT_BLOG";
export const DELETE_BLOG: string = "DELETE_BLOG";
export const VIEW_BLOG: string = "VIEW_BLOG";

export const addBlog = (blog: IBlog) => {
    return {
        type: ADD_BLOG,
        blog: {
            ...blog,
            id: uuid.v1(),
            date: moment(),
        },
    }
}

export const deleteBlog = (blogId: string) => {
    return {
        type: DELETE_BLOG,
        blogId: blogId,
    }
}

export const ediBlog = (blog: IBlog) => {
    return {
        type: EDIT_BLOG,
        blog,
    }
}




