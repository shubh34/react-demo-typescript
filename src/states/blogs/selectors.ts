import moment  from "moment";
import { IRootState } from "../../store";

import { dateFormat, maxContentCharacterLimit } from "../../configs/config";

export const getBlogList = (state: IRootState) => 
    {
        return state.blogListing.blogs.map(blog => {
            return {
                ...blog,
                date: moment(blog.date).format(dateFormat),
                readMore: blog.content.length > maxContentCharacterLimit,
                content: blog.content.substring(0, maxContentCharacterLimit),
            };
        });
    }
export const hasBlogId = (state: IRootState, id: string)  => state.blogListing.blogs.some(blog => blog.id === id);

export const getBlogById = (state: IRootState, id: string) => {
    {
        const blog = state.blogListing.blogs.find(blog => blog.id === id);
        if(blog) {
            return  {
                ...blog,
                date: moment(blog.date).format(dateFormat),
            }
        }
        return blog;
    }
}