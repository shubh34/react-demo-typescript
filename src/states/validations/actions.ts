
export const INIT_VALIDATION: string = "INIT_VALIDATION";
export const UPDATE_VALIDATION: string = "UPDATE_VALIDATION";
export const RESET_VALIDATION: string = "RESET_VALIDATION";

export const initValidation = (name: string, isValid: boolean) => {
    return {
        type: INIT_VALIDATION,
        field: {
            name,
            isValid,
        }
    }
}

export const updateValidation = (name: string, isValid: boolean) => {
    return {
        type: UPDATE_VALIDATION,
        field: {
            name,
            isValid,
        }
    }
}
export const resetValidation = () => {
    return {
        type: RESET_VALIDATION,
    }
}


