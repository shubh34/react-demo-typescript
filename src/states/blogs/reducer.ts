import { IBlogsState } from "./types";
import { ADD_BLOG } from "./action";

const init: IBlogsState = {
    blogs: []
};


const reducer = (state = init, action: any) : IBlogsState => {
	switch (action.type) {
	case ADD_BLOG: 
		return {
			...state,
			blogs: [...state.blogs, action.blog],
		}
	default:
		console.log('state', state);
		return state;
	}
};
export default reducer;
