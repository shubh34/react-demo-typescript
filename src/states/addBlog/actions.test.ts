import { IAddBlogState } from './types';
import { INIT_ADD_BLOG, UPDATE_FORM_FIELD, ADD_BLOG_CATEGORY, DELETE_BLOG_CATEGORY, initAddBlogForm, updateFormField, addCategory, deleteCategory } from './actions';
import { addBlog } from '../blogs/action';

Date.now = jest.fn(() => 1487076708000);
jest.mock('node-uuid', () => ({ v1: jest.fn(() => 'uuid') }));

describe('addBlog action', () => {
    it('should create addBlog action', () => {
        expect(initAddBlogForm()).toEqual({
            type: INIT_ADD_BLOG,
        });
    });
    it('should create updateFormField action', () => {
        expect( updateFormField('title', 'My Blog')).toEqual({
            type: UPDATE_FORM_FIELD,
            field: {
                value: "My Blog",
                name: "title",
            },
        });
    });
    it('should create an action to add blog', () => {
        const addBlogObject: IAddBlogState = {
            title: 'My Blog',
            content: 'My Content',
            categories: ['2323', '23213'],
        };
        expect(addBlog(addBlogObject)).toMatchSnapshot();
    });
    it('should create an action to add category', () => {
      
        expect(addCategory('test')).toEqual({
            type: ADD_BLOG_CATEGORY,
            category: 'test',
        });
    });

    it('should create an action to delete category', () => {
      
        expect(deleteCategory('test')).toEqual({
            type: DELETE_BLOG_CATEGORY,
            category: 'test',
        });
    });
    
});