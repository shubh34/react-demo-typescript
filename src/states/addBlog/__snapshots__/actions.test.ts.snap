// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addBlog action should create an action to add blog 1`] = `
Object {
  "blog": Object {
    "categories": Array [
      "2323",
      "23213",
    ],
    "content": "My Content",
    "date": "2017-02-14T12:51:48.000Z",
    "id": "uuid",
    "title": "My Blog",
  },
  "type": "ADD_BLOG",
}
`;
